### PR TITLE
Build: Make ABACUS-PW.x able to be compiled without LCAO 

### PR DESCRIPTION
**Pull request by Qianruipku**
1. It is compiled with Makefile in src_pw. As a result, we do not need elpa or cereal libs to do DFT calculations with pw basis.
2. delete redudant tests in src_pw since hslover have similar tests
3. add tests to compile abacus_pw.exe in src_pw
It is a temporary treatment. Maybe in the future ABACUS can compile ABACUS-PW.x with cmake directly but not with camke -DBUILD_TESTING=ON.

### CODE REVIEW

**Review by dyzheng — COMMENTED** [review]
"source/src_pw/test/pw.cpp" is an empty file?

**Review by dyzheng — APPROVED** [review]

**Comment by Qianruipku** [comment]
> "source/src_pw/test/pw.cpp" is an empty file?

Sorry, I will delete it.

File: source/src_pw/test/CMakeLists.txt [hunk 1]
```diff
@@ -1,30 +1,62 @@
...
-  SOURCES diago_david_test.cpp ../diago_david_old.cpp ../../src_parallel/parallel_reduce.cpp 
-)
 
 AddTest(
-  TARGET hsolver_cg
-  LIBS ${math_libs} base 
-  SOURCES diago_cg_test.cpp ../diago_cg_old.cpp ../../src_parallel/parallel_reduce.cpp 
+  TARGET abacus_pw.exe #Test PW ABACUS without LCAO #Please also update Makefile in src_pw
+  LIBS ${math_libs} planewave base surchem xc_ operator_ks_pw neighbor symmetry
+  SOURCES ../../driver.cpp ../../input.cpp ../../input_conv.cpp ../../driver_run.cpp ../../main.cpp ../../input_update.cpp
```
Comment (caic99) [inline source/src_pw/test/CMakeLists.txt:9]: What's the point adding source files one by one, but not importing whole library target? Will this method cause compilation error?

Comment (Qianruipku) [reply source/src_pw/test/CMakeLists.txt:9]: Because the whole library target is compiled with __LCAO and it can not be used.

Comment (caic99) [reply source/src_pw/test/CMakeLists.txt:9]: I suggest add full support to build planewave module in cmake, and this test will be more elegant. What do you think?

Comment (Qianruipku) [reply source/src_pw/test/CMakeLists.txt:9]: I agree and I suggest we can define a new option "ONLY_PW". Here, we do not need prepare elpa or cereal to compile ABACUS-PW.  And a new test workflow in github should be added.

**Comment by caic99** [comment]
Hi @Qianruipku ,

I don't think this workaround is appropriate compiling all source files for plane-wave basis again to generate one executable. There is an option of `__LCAO`; maybe it can not works well right now, but I suggest you fixing it before adding this test.

Besides, the test you added does not take a input file. Does it go with your plan? See [log](https://github.com/deepmodeling/abacus-develop/actions/runs/3217630209/jobs/5260830509#step:5:3864).

**Pull request merged**